### PR TITLE
node:  fix CLI

### PR DESCRIPTION
**Pull request by deepfire**
1. Fix the `cardano-node` CLI to properly report `--socket-path` as mandatory.
2. Small non-semantic style changes to better conform with the style guide.
3. Fix the `cardano-node` CLI to list commands when none are specified.

### CODE REVIEW

**Comment by erikd** [comment]
I have just checked out this branch and had a look at the help output of `cardano-node`. I still think the help output code be very much better and in fact should be much better before we have non-IOHK users running this.

It would really help if you could list all the different ways this program could be run.

File: cardano-config/src/Cardano/Config/CommonCLI.hs [hunk 1]
```diff
@@ -135,6 +135,12 @@ lastStrOption args = Last <$> optional (strOption args)
 lastFlag :: a -> a -> Mod FlagFields a -> Parser (Last a)
 lastFlag def act opts  = Last <$> optional (flag def act opts)
 
+-- | Mandatory versions of option parsers.
+--   Use these for the cases when presets don't define a default value
+--   for a particular field -- i.e. when the field is set to 'mempty'.
+lastStrOptionM :: IsString a => Mod OptionFields a -> Parser (Last a)
+lastStrOptionM args = Last . Just <$> strOption args
```
[the remark this answers is not in the document]
Comment (deepfire) [reply cardano-config/src/Cardano/Config/CommonCLI.hs]: @erikd, the mass-use of `Last` is intentional -- the idea is to be able to transparently merge layers of configuration at the top level:

1. presets
2. configuration files, and
3. CLI args

File: cardano-config/src/Cardano/Config/CommonCLI.hs [hunk 2]
```diff
@@ -21,7 +21,7 @@ module Cardano.Config.CommonCLI
   ) where
 
 import           Cardano.Prelude hiding (option)
-import           Prelude
+import           Prelude hiding ((.))
```
[the remark this answers is not in the document]
Comment (deepfire) [reply cardano-config/src/Cardano/Config/CommonCLI.hs]: Done!

**Comment by deepfire** [comment]
> I have just checked out this branch and had a look at the help output of cardano-node. I still think the help output code be very much better and in fact should be much better before we have non-IOHK users running this.
>
> It would really help if you could list all the different ways this program could be run.

I think we should separate the trace acceptor, it doesn't seem to belong.

That would allow things to be streamlined..

**Comment by deepfire** [comment]
@erikd, I did:

1. The separation of `trace-acceptor` -- and now the `cardano-node`'s CLI is much clearer,
2. Verify that all args that are listed as required are exactly those that are required.

Please consider that f.e. `(-n|--node-id NODE-ID)` is a mandatory arg, despite being printed treacherously similarly to square-bracketed optional args.

**Comment by erikd** [comment]
The help output really needs improving. I think all the tracing options should be hidden unless the user specificially asks for help with the tracing options.

It would still like to see the actual command like you expect to use (ignoring tracing options).

**Comment by deepfire** [comment]
@erikd, found out a rather sneaky way to hide the tracing options, without losing almost anything.

I didn't make the code itself pretty -- let's agree on the help output before I do it : -)

A series of exhibits:

1. default help:
```
Missing: --node-id NODE-ID --topology FILEPATH --host-addr HOST-NAME --port PORT
(--byron-legacy | --bft | --praos | --mock-pbft | --real-pbft)
--log-config LOGCONFIG

Usage: cardano-node --node-id NODE-ID --topology FILEPATH --host-addr HOST-NAME
                    --port PORT (--byron-legacy | --bft | --praos | --mock-pbft
                    | --real-pbft) [--live-view] --log-config LOGCONFIG
                    [--database-path FILEPATH] [--genesis-file FILEPATH]
                    [--genesis-hash GENESIS-HASH]
                    [--pbft-signature-threshold DOUBLE]
                    [--delegation-certificate FILEPATH] [--signing-key FILEPATH]
                    [--require-network-magic] [--slot-duration SECONDS]
                    [--socket-dir FILEPATH] [--tracing-off]
                    ([--tracing-verbosity-minimal] |
                    [--tracing-verbosity-maximal] |
                    [--tracing-verbosity-normal]) [--help] [--help-tracing]
  Start node of the Cardano blockchain.
```

2. With --help:
```
Usage: cardano-node --node-id NODE-ID --topology FILEPATH --host-addr HOST-NAME
                    --port PORT (--byron-legacy | --bft | --praos | --mock-pbft
                    | --real-pbft) [--live-view] --log-config LOGCONFIG
                    [--database-path FILEPATH] [--genesis-file FILEPATH]
                    [--genesis-hash GENESIS-HASH]
                    [--pbft-signature-threshold DOUBLE]
                    [--delegation-certificate FILEPATH] [--signing-key FILEPATH]
                    [--require-network-magic] [--slot-duration SECONDS]
                    [--socket-dir FILEPATH]

  --node-id NODE-ID        PBFT node ID to assume.
  --topology FILEPATH      The path to a file describing the topology.
  --host-addr HOST-NAME    The ipv6 or ipv4 address
  --port PORT              The port number
  --byron-legacy           Byron/Ouroboros Classic suite of algorithms
  --bft                    BFT consensus
  --praos                  Praos consensus
  --mock-pbft              Permissive BFT consensus with a mock ledger
  --real-pbft              Permissive BFT consensus with a real ledger
  --live-view              Live view with TUI.
  --log-config LOGCONFIG   Configuration file for logging
  --database-path FILEPATH Directory where the state is stored.
  --genesis-file FILEPATH  The filepath to the genesis file.
  --genesis-hash GENESIS-HASH
                           The genesis hash value.
  --pbft-signature-threshold DOUBLE
                           The PBFT signature threshold.
  --delegation-certificate FILEPATH
                           Path to the delegation certificate.
  --signing-key FILEPATH   Path to the signing key.
  --require-network-magic  Require network magic in transactions.
  --slot-duration SECONDS  The slot duration (seconds)
  --socket-dir FILEPATH    Directory with local sockets:
                           ${dir}/node-{core,relay}-${node-id}.socket
```

3. With --help-tracing:
```
Additional tracing options:

  --tracing-off            Tracing globally turned off.
  --tracing-verbosity-minimal
                           Minimal level of the rendering of captured items
  --tracing-verbosity-maximal
                           Maximal level of the rendering of captured items
  --tracing-verbosity-normal
                           the default level of the rendering of captured items
  --trace-chain-db         Verbose tracer of ChainDB.
  --trace-chain-sync-client
                           Trace ChainSync client.
  --trace-chain-sync-header-server
                           Trace ChainSync server (headers).
  --trace-chain-sync-block-server
                           Trace ChainSync server (blocks).
  --trace-block-fetch-decisions
                           Trace BlockFetch decisions made by the BlockFetch
                           client.
  --trace-block-fetch-client
                           Trace BlockFetch client.
  --trace-block-fetch-server
                           Trace BlockFetch server.
  --trace-tx-inbound       Trace TxSubmission server (inbound transactions).
  --trace-tx-outbound      Trace TxSubmission client (outbound transactions).
  --trace-local-tx-submission-server
                           Trace local TxSubmission server.
  --trace-mempool          Trace mempool.
  --trace-forge            Trace block forging.
  --trace-chain-sync-protocol
                           Trace ChainSync protocol messages.
  --trace-block-fetch-protocol
                           Trace BlockFetch protocol messages.
  --trace-tx-submission-protocol
                           Trace TxSubmission protocol messages.
  --trace-local-chain-sync-protocol
                           Trace local ChainSync protocol messages.
  --trace-local-tx-submission-protocol
                           Trace local TxSubmission protocol messages.
  --trace-ip-subscription  Trace IP Subscription messages.
  --trace-dns-subscription Trace DNS Subscription messages.
  --trace-dns-resolver     Trace DNS Resolver messages.
```

**Pull request merged**